Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -650,12 +640,19 @@ protected Sheet createSheet() {
                     NO_DESCR,
                     actualMimeType));
         }
+        
+        BlackboardArtifact.Type artifactType = null;
+        try {
+            artifactType = artifact.getType();
```

Comment:
For custom artifact types this is adding a DB call onto the EDT.  I know that its an edge case but it is a potential performance issue.  While we are trying to sort out other issues it doesn't make sense to add extra DB calls.

Reply:
Sure.  Did you have something in mind for how to fix this?  Do you want to move the call into the constructor with `srcContent.getUniquePath()` and have an instance field with the category?

Reply:
The constructor is also on the EDT.

Reply:
Sure.  I believe the call to `srcContent.getUniquePath()` currently in the constructor could potentially be making a database call (at least for `AbstractFile`) as well.  We could move the potential db call for the artifact type into the constructor so that whenever we fix the db/EDT issue for `BlackboardArtifactNode`, these two calls will be in the same place.  Did you want to do that or did you have something else in mind?

Reply:
The artifact type is not going to change so having this call in createSheet doesn't make sense.  Your best option, for now, is to cache the type on construction.

Reply:
Sounds good.  Will do.